docs/howtos/model_surgery.rst

```diff
@@ -0,0 +1,232 @@
+Model Surgery
+==============================
+
+.. testsetup::
+
+  import functools
+  import numpy as np
+  import jax
+  from jax import lax, random, numpy as jnp
+  import flax
+  from flax import optim, traverse_util
+
+  from flax import linen as nn
+  from flax.core import unfreeze, freeze
+
+We will show how to get a flat dict of all the tensors, and then go back to a 
+nested, frozen dict. We show this for both Flax modules and optimizers.
+
+Surgery with Flax Modules
+--------------------------------
+
+Let's first make a little model for our demo.
```

Comment:
Suggestion:

- For those use who have knowledge of numerical computing (NumPy) but not as familiar with NNs, I propose to maybe rephrase this like so:

```suggestion
Let's create a small convolutional neural network model for our demo.
```